akd/src/append_only_zks.rs

```diff
@@ -523,31 +523,27 @@ impl Azks {
         &self,
         storage: &S,
     ) -> Result<H::Digest, AkdError> {
-        self.get_root_hash_at_epoch::<_, H>(storage, self.get_latest_epoch())
+        self.get_root_hash_safe::<_, H>(storage, self.get_latest_epoch())
             .await
     }
 
-    /// Gets the root hash of the tree at a epoch.
-    /// Since this is accessing the root node and the root node exists at all epochs that
-    /// the azks does, this would never be called at an epoch before the birth of the root node.
-    pub async fn get_root_hash_at_epoch<S: Storage + Sync + Send, H: Hasher>(
+    /// Gets the root hash of the tree at the latest epoch if the passed epoch
+    /// is equal to the latest epoch. Will return an error otherwise.
+    pub async fn get_root_hash_safe<S: Storage + Sync + Send, H: Hasher>(
```

Comment:
Should we just make this `get_root_hash` and not take an epoch at all? I know it's a bigger refactoring, but if there's only 1 valid value of `epoch` we support, and we can get it from `self.latest_epoch` then perhaps we shouldn't take it as an arg at all...?

Reply:
It's more for the caller to validate that the root hash being fetched is at the epoch that the caller expects. Say for example the caller is in a state where it thinks the epoch is 5, and wants to get the root hash for epoch 5. However the directory is already at epoch 6.

With the regular `get_root_hash`function, the caller would get the root hash for epoch 6 and treat it as the root hash for epoch 5. With this API the directory can return an Err and trigger any recovery mechanisms for inconsistent state. Does that make sense?

Reply:
Yeah OK that makes sense. Thanks!